include/xtensor/xshape.hpp

```diff
@@ -0,0 +1,687 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay and Wolf Vollprecht    *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_XSHAPE_HPP
+#define XTENSOR_XSHAPE_HPP
+
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <cstdlib>
+#include <cstring>
+#include <initializer_list>
+#include <iterator>
+#include <memory>
+
+#include "xexception.hpp"
+#include "xstorage.hpp"
+
+#if (__GNUC__ || __clang__)
+    #define likely(x)       __builtin_expect((x), 1)
+    #define unlikely(x)     __builtin_expect((x), 0)
```

Comment:
Does it really improve performances? Else I'd be in favor of removing that definitions

Reply:
I think you're right and it can be removed. I copied this from LLVM's small vector and thought it was cool :)